core/frontend/src/render/webgl/glsl/Clipping.ts

```diff
@@ -99,13 +99,13 @@ const applyClipPlanesPostlude = `
 
   numSetsClippedBy += int(clippedByCurrentPlaneSet);
   if (numSetsClippedBy == numPlaneSets) {
-    if (u_outsideRgba.a > 0.0) {
+    if (!u_overrideClipStyle && (u_outsideRgba.a > 0.0)) {
```

Comment:
What about ClipIntersectionStyle?

Reply:
My initial testing didn't show that ClipIntersectionStyle was part of the related bug, but I believe that was because I limited the width so it wasn't noticable. After a little more testing with a wider width just now, I see the ClipIntersectionStyle is being applied to the map tiles, so I'll work on fixing this.